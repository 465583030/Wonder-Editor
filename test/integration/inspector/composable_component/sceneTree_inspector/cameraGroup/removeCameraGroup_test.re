open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("test remove camera group", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initStateWithJob(
        ~sandbox,
        ~noWorkerJobRecord=
          NoWorkerJobConfigToolEngine.buildNoWorkerJobConfig(
            ~loopPipelines=
              {|
                   [
                       {
                           "name": "default",
                           "jobs": [
                               {
                                   "name": "dispose"
                               }
                           ]
                       }
                   ]
               |},
            (),
          ),
        (),
      );

      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setSceneFirstCameraToBeCurrentSceneTreeNode,
      );

      CurrentSelectSourceEditorService.setCurrentSelectSource(
        EditorType.SceneTree,
      )
      |> StateLogicService.getAndSetEditorState;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test remove all scene->camera groups", () => {
      test("should remove from inspector", () => {
        SceneTreeNodeDomTool.OperateDefaultScene.getCameraGroupFromCamera()
        |> OperateComponentEventTool.removeComponentFromCurrentGameObject;

        BuildComponentTool.buildInspectorComponent(
          TestTool.buildEmptyAppState(),
          InspectorTool.buildFakeAllShowComponentConfig(),
        )
        |> ReactTestTool.createSnapshotAndMatch;
      });
      test("should remove activedBasicCameraView from editorState", () => {
        SceneTreeNodeDomTool.OperateDefaultScene.getCameraGroupFromCamera()
        |> OperateComponentEventTool.removeComponentFromCurrentGameObject;

        GameViewEditorService.getActivedBasicCameraView(
          StateEditorService.getState(),
        )
        |> expect == None;
      });
    });

    describe("test still has other camera groups after remove ", () => {
      beforeEach(() => {
        HeaderTool.triggerAddBox();

        SceneTreeNodeDomTool.OperateDefaultScene.getNewGameObjectDomIndex()
        |> SceneTreeTool.clearCurrentGameObjectAndSetTreeSpecificGameObject;

        AddableComponentTool.addCameraGroupInBox();
      });

      test("should mark last scene camera to be active", () => {
        let lastBasicCameraView =
          SceneEditorService.getCurrentSceneTreeNode(
            StateEditorService.getState(),
          )
          |> OptionService.unsafeGet
          |> GameObjectComponentEngineService.getBasicCameraViewComponent(
               _,
               StateEngineService.unsafeGetState(),
             );
        MainEditorSceneTool.setSceneFirstCameraToBeCurrentSceneTreeNode();

        SceneTreeNodeDomTool.OperateDefaultScene.getCameraGroupFromCamera()
        |> OperateComponentEventTool.removeComponentFromCurrentGameObject;

        GameViewEditorService.getActivedBasicCameraView(
          StateEditorService.getState(),
        )
        |> expect == Some(lastBasicCameraView);
      });
    });
  });