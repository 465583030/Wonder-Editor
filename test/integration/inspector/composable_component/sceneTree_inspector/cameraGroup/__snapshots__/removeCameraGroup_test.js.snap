// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test remove camera group test has no camera group after remove  test remove current camera group should remove from inspector 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div>
      <article
        className="wonder-string-input"
      >
        <span
          className="component-label"
        >
          Name : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="camera"
        />
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            position : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="40"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            rotation : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            scale : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
          </article>
        </div>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <button
          className="addable-btn"
          onClick={[Function]}
        >
          Add Component
        </button>
      </div>
    </article>
  </article>
</article>
`;

exports[`test remove camera group test still has other camera groups after remove  test remove cameraGroup component, should remove from inspector 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div>
      <article
        className="wonder-string-input"
      >
        <span
          className="component-label"
        >
          Name : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="camera"
        />
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            position : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="40"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            rotation : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            scale : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
          </article>
        </div>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <button
          className="addable-btn"
          onClick={[Function]}
        >
          Add Component
        </button>
      </div>
    </article>
  </article>
</article>
`;
